### PR TITLE
Android 12 fixes

### LINKED ISSUE #410: Android 12 (S) support
Android 12 has been in developer preview since late February, with the final release probably in September or October.  Need to make sure libraries are all compatible and the testapp works.  A summary of changes is here:  https://developer.android.com/about/versions/12/summary and apps that target Android 12:  https://developer.android.com/about/versions/12/behavior-changes-12.

So far I've only noticed that the testapp's manifest needs to be updated to explicitly declare the `android:exported` attribute for activities, services and broadcast receivers. `R2DispatcherActivity` and `MainActivity` (once #403 is merged) both need exported to be set to true, while the others can be false.

### PR DESCRIPTION
**Pull request by stevenzeck**
Two fixes for getting it running on an Android 12 device:

1. Marking activities with intent-filters with `android:exported`
2. Removing `InputMethodManager` `toggleSoftInput` with `showSoftInput`

A few minor things:

1. Having `dataBinding = true` instead of `dataBinding true` generates warnings, so switched that

Closes #410

### CODE REVIEW

**Comment by stevenzeck** [comment]
I don't know why `jcenter()` in the testapp's gradle file is needed for the navigator dependency in order for the GitHub Actions build to pass.  Another reason to have a monorepo.

File: r2-testapp/src/main/java/org/readium/r2/testapp/reader/EpubReaderFragment.kt [hunk 1]
```diff
@@ -236,9 +236,8 @@ class EpubReaderFragment : VisualReaderFragment(), EpubNavigatorFragment.Listene
             model.cancelSearch()
             menuSearchView.setQuery("", false)
 
-            (activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager)?.toggleSoftInput(
-                InputMethodManager.SHOW_FORCED,
-                InputMethodManager.HIDE_IMPLICIT_ONLY
+            (activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager)?.showSoftInput(
+                this.view, InputMethodManager.SHOW_FORCED
```
Comment (qnga) [inline r2-testapp/src/main/java/org/readium/r2/testapp/reader/EpubReaderFragment.kt:240]: Don't  we want to hide the keyboard here instead of showing it?

Comment (stevenzeck) [reply r2-testapp/src/main/java/org/readium/r2/testapp/reader/EpubReaderFragment.kt:240]: Nope, this function clears out both the result list (if present) and whatever text is in the search box, then focuses back into the search box ready for the user to do a new search.  So we want to show the keyboard at this point.

**Review by mickael-menu — APPROVED** [review]
Thanks for taking the time to tackle Android 12 👍

**Pull request merged**